packages/headless/src/features/triggers/triggers-slice.ts

```diff
@@ -4,18 +4,32 @@ import {getTriggerInitialState} from './triggers-state';
 import {
   Trigger,
   TriggerRedirect,
-  isTriggerRedirect,
+  isRedirectTrigger,
+  isQueryTrigger,
+  TriggerQuery,
 } from './../../api/search/trigger';
+import {clearQueryTrigger} from './trigger-actions';
 
 export const triggerReducer = createReducer(
   getTriggerInitialState(),
   (builder) =>
-    builder.addCase(executeSearch.fulfilled, (state, action) => {
-      const redirectTriggers: Trigger[] = action.payload.response.triggers.filter(
-        (trigger) => isTriggerRedirect(trigger)
-      );
-      state.redirectTo = redirectTriggers.length
-        ? (redirectTriggers[0] as TriggerRedirect).content
-        : '';
-    })
+    builder
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        const redirectTriggers: Trigger[] = action.payload.response.triggers.filter(
+          (trigger) => isRedirectTrigger(trigger)
+        );
+        state.redirectTo = redirectTriggers.length
+          ? (redirectTriggers[0] as TriggerRedirect).content
+          : '';
+
+        const queryTriggers: Trigger[] = action.payload.response.triggers.filter(
+          (trigger) => isQueryTrigger(trigger)
+        );
+        state.query = queryTriggers.length
+          ? (queryTriggers[0] as TriggerQuery).content
+          : '';
```

Comment:
Have you tested with the fetchMoreResults functionality? I suspect you may have to clear the query on fetchMoreResults.fulfilled.